### PR TITLE
Avoid NaN when converting screen coordinates to geographical coordinates

### PR DESCRIPTION
**Pull request by tobrun**
### Summary of changes
This patch a crash reported by a user. The crash is very much edge case and while the usage of ScreenCoordinate(0,0) might be a band-aid solution, it's better than crashing the end-user application. 

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

### CODE REVIEW

**Review by Chaoba — APPROVED** [review]

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 1]
```diff
@@ -1787,6 +1787,16 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase, MapStyleObserve
   }
 }
 
+/**
+ * Sanitize ScreenCoordinate to avoid passing NaN to core
+ */
+private fun ScreenCoordinate.sanitizeNaN(): ScreenCoordinate {
+  return ScreenCoordinate(
+    if (!x.isNaN()) x else 0.0,
```
Comment (yunikkk) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1795]: Log an error if it's nan?

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 2]
```diff
@@ -1393,7 +1393,7 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase, MapStyleObserve
     // Prevent drag start in area around horizon to avoid sharp map movements
     val topMapMargin = 0.04 * mapTransformDelegate.getSize().height
     val reprojectErrorMargin = min(10.0, topMapMargin / 2)
-    val point = ScreenCoordinate(pixel.x, pixel.y - topMapMargin)
+    val point = ScreenCoordinate(pixel.x, pixel.y - topMapMargin).sanitizeNaN()
```
Comment (yunikkk) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1396]: Seems like we're hiding the actual error here, with either `topMapMargin` or `pixel` holding the original NaN.
Maybe sanitize pixel / margin independently, will at least log the wrong one?

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 3]
```diff
@@ -1787,6 +1787,16 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase, MapStyleObserve
   }
 }
 
+/**
+ * Sanitize ScreenCoordinate to avoid passing NaN to core
+ */
+private fun ScreenCoordinate.sanitizeNaN(): ScreenCoordinate {
+  return ScreenCoordinate(
+    if (!x.isNaN()) x else 0.0,
+    if (!y.isNaN()) x else 0.0
```
Comment (alexander-kulikovskii) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1796]:
```suggestion
    if (!y.isNaN()) y else 0.0
```

Comment (alexander-kulikovskii) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1796]:
```suggestion
    x = if (!x.isNaN()) x else 0.0,
    y = if (!y.isNaN()) x else 0.0
```

Comment (tobrun) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1796]: ha yes :D need to add tests

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt (hunk 2 above)
Comment (tobrun) [reply plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt]: from stacktrace I'm seeing that both values from ScreenCoordinate are NaN

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 4]
```diff
@@ -1393,7 +1395,22 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase, MapStyleObserve
     // Prevent drag start in area around horizon to avoid sharp map movements
     val topMapMargin = 0.04 * mapTransformDelegate.getSize().height
     val reprojectErrorMargin = min(10.0, topMapMargin / 2)
-    val point = ScreenCoordinate(pixel.x, pixel.y - topMapMargin)
+
+    // sanitize input x to be non NaN
+    var pointX = pixel.x
+    if (pointX.isNaN()) {
+      Log.e(TAG, "isPointAboveHorizon: screen coordinate x is NaN.")
```
Comment (alexander-kulikovskii) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1402]:
```suggestion
      logE(TAG, "isPointAboveHorizon: screen coordinate x is NaN.")
```

Comment (alexander-kulikovskii) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1402]: nit:
maybe add tests for logger?

File: plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt [hunk 5]
```diff
@@ -1393,7 +1395,22 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase, MapStyleObserve
...
+      pointX = 0.0
+    }
+
+    // sanitize input y to be non NaN
+    var pointY = pixel.y
+    if (pointY.isNaN()) {
+      Log.e(TAG, "isPointAboveHorizon: screen coordinate y is NaN.")
+      pointY = 0.0
+    }
```
Comment (alexander-kulikovskii) [inline plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt:1411]:
```suggestion
    val pointY = if (pointY.isNaN()) {
      Log.e(TAG, "isPointAboveHorizon: screen coordinate y is NaN.")
      0.0
    } else {
        pixel.y
    }
```

**Review by alexander-kulikovskii — APPROVED** [review]

**Pull request merged**